pkg/resmap/resmap.go

```diff
@@ -587,3 +593,49 @@ func (m *resWrangler) appendReplaceOrMerge(
 	}
 	return nil
 }
+
+// Select returns a list of resources that
+// are selected by a Selector
+func (m *resWrangler) Select(s types.Selector) ([]*resource.Resource, error) {
+	ns := regexp.MustCompile(s.Namespace)
+	nm := regexp.MustCompile(s.Name)
+	var result []*resource.Resource
+	for _, r := range m.Resources() {
+		// matches the namespace when namespace is not empty in the selector
+		matched := ns.MatchString(r.GetNamespace())
+		if r.GetNamespace() != "" && !matched {
```

Comment:
- confusing that `matched` is computed when it might not be needed

- With respect to namespaces and names, it's best to explicitly consult `r.CurId().Name` or `r.OrgId().Name` instead of going to the resource itself. Then it's clear if we are hunting by original name or new name. Patch targeting tries the original name first, then current name, to allow the user to specify either in the kustomization file.

- likewise it might be best to match the namespace field against `id.EffectiveNamespace`.  this conflates the empty and "default" namespace.

- it should be an error somewhere if someone makes a Selector that specifies a non-empty namespace for a non - NamespaceableKind()  - but maybe that's a different PR

Reply:
i'm also trying to future proof this against Kind changes (e.g. a patch that converts a Deployment to a Job).  The notions of CurId and OrgId should support that.  Ditto namespace edits, apiVersion, etc.

Reply:
> I'm also trying to future proof this against Kind changes (e.g. a patch that converts a Deployment to a Job). The notions of CurId and OrgId should support that. Ditto namespace edits, apiVersion, etc.

This can be done in a different PR.